dbt/include/spark/macros/materializations/snapshot.sql

```diff
@@ -135,7 +135,9 @@
 
   {% else %}
 
-      {{ adapter.valid_snapshot_target(target_relation) }}
+      {% set stcn = config.get("snapshot_table_column_names") or get_snapshot_table_column_names() %}
+
+      {{ adapter.valid_snapshot_target(target_relation, stcn) }}
```

Comment:
Calling `adapter.valid_snapshot_target` with the new second parameter will break unless we use the new version of `dbt-adapters`. We need to bump the minor on `dbt-adapters` and then bump the pin in `dbt-spark` to this new minor.